packages/mcap-support/src/parseFlatbufferSchema.ts

```diff
@@ -0,0 +1,202 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import { ByteBuffer } from "flatbuffers";
+import { BaseType, Schema, SchemaT, FieldT, Parser, Table } from "flatbuffers_reflection";
+
+import { RosMsgField } from "@foxglove/rosmsg";
+
+import { RosDatatypes } from "./types";
+
+function typeForSimpleField(type: BaseType): string {
+  switch (type) {
+    case BaseType.Bool:
+      return "bool";
+    case BaseType.Byte:
+      return "int8";
+    case BaseType.UType:
+    case BaseType.UByte:
+      return "uint8";
+    case BaseType.Short:
+      return "int16";
+    case BaseType.UShort:
+      return "uint16";
+    case BaseType.Int:
+      return "int32";
+    case BaseType.UInt:
+      return "uint32";
+    case BaseType.Long:
+      return "int64";
+    case BaseType.ULong:
+      return "uint64";
+    case BaseType.Float:
+      return "float32";
+    case BaseType.Double:
+      return "float64";
+    case BaseType.String:
+      return "string";
+    case BaseType.Vector:
+    case BaseType.Obj:
+    case BaseType.Union:
+    case BaseType.Array:
+      return "";
+    case BaseType.None:
+    case BaseType.MaxBaseType:
+      throw new Error("None is not a valid type.");
+  }
+  return "";
+}
+
+// eslint-disable-next-line no-restricted-syntax
+function flatbufferString(unchecked: string | Uint8Array | null | undefined): string {
+  if (typeof unchecked === "string") {
+    return unchecked;
+  }
+  throw new Error("Not a string.");
```

Comment:
More helpful error message might be nice, e.g.
```suggestion
  throw new Error(`Expected string, found ${typeof unchecked}`);
```